src/compiler/crystal/command.cr

```diff
@@ -346,6 +346,11 @@ class Crystal::Command
         @error_trace = true
       end
 
+      opts.on("--nodefaultlibs", "Disable standard library locations") do
+        compiler.flags << "-nodefaultlibs"
```

Comment:
Is this flag necessary?

Reply:
In most cases, if we're trying to prevent the compiler from using default library paths, I would say yes. GCC for example will try and use the standard system libraries if this is not added as a flag. This goes slightly beyond my initial intention of taking out the hard-coded path by following the change through down to the linker.

Reply:
This is setting `-D` flags for crystal, not linker flags.